ofrak_patch_maker/ofrak_patch_maker/binary_parser/gnu.py

```diff
@@ -30,20 +31,19 @@ class GNU_ELF_Parser(AbstractBinaryFileParser):
         flags=re.MULTILINE,
     )
 
-    def parse_symbols(self, output: str) -> Dict[str, int]:
+    def parse_symbols(self, output: str) -> Dict[str, Tuple[int, LinkableSymbolType]]:
         """
-        Use `objdump` with the `--syms` flag to get info on all symbols in a file. Parses columns
-        based on: <https://stackoverflow.com/a/16471895/16690095>.
+        Use `objdump` with the `--syms` flag to get info on all defined symbols in a file. Parses
+        columns based on: <https://stackoverflow.com/a/16471895/16690095>.
         """
         result = {}
-        for symbol_data in self._re_symbol_prog.finditer(output):
-            name = symbol_data.group("name")
-            addr = symbol_data.group("address")
-            symbol_section = symbol_data.group("section")
-
-            if name and addr:
-                if symbol_section and symbol_section != "*UND*":
-                    result[name] = int(addr, 16)
+        symbols = self._get_all_symbols(output)
+        for symbol in symbols:
+            if symbol[2] != "*UND*":
+                if "F" in symbol[3]:
+                    result[symbol[0]] = (symbol[1], LinkableSymbolType.FUNC)
+                else:
+                    result[symbol[0]] = (symbol[1], LinkableSymbolType.UNDEF)
```

Comment:
Unpacking the tuple when iterating over it can be much more readable than indexing into it:

```suggestion
        symbols = self._get_all_symbols(output)
        for sym_name, sym_vaddr, sym_section, sym_type in symbols:
            if sym_section != "*UND*":
                if "F" in sym_type:
                    result[sym_name] = (sym_vaddr, LinkableSymbolType.FUNC)
                else:
                    # TODO handle data symbols and distinguish between RO and RW symbols with section info
                    result[sym_name] = (sym_vaddr, LinkableSymbolType.UNDEF)
```

I also suggested adding a TODO wrt data symbols that aren't handled at the moment. UNDEF type symbols won't be turned into stubs (or even `LinkableSymbol` resources in the `define_linkable_symbols_from_patch` method. For simpler use cases (what we expect to use these injected symbols for), this is probably fine.